JustSaying.IntegrationTests/Fluent/IntegrationTestBase.cs

```diff
@@ -90,7 +90,7 @@ protected async Task WhenAsync(IServiceCollection services, Func<IMessagePublish
             IMessagePublisher publisher = serviceProvider.GetRequiredService<IMessagePublisher>();
             IMessagingBus listener = serviceProvider.GetRequiredService<IMessagingBus>();
 
-            using (var source = new CancellationTokenSource(Timeout))
+            using (var source = new CancellationTokenSource())
```

Comment:
This is not easy to reason about. Should it not cancel at some point? If not on a timeout upfront, then maybe after the `.WhenAny` - assuming that if `actionTask` goes wrong, then the `delayTask` will complete first?

Why do we do `Task.Delay(Timeout, source.Token)` rather than `Task.Delay(Timeout)` ? [The docs suggest](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay?view=netframework-4.7.2#System_Threading_Tasks_Task_Delay_System_TimeSpan_System_Threading_CancellationToken_) that the first overload "observes" the token. But it won't be signalled, will it?

Reply:
Thanks, I've messed this up for sure. As far as I can see, there are 2 patterns and we are straddling them. I'll push an update and ping you.